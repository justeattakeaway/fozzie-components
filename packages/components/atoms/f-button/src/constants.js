// eslint-disable-next-line import/prefer-default-export
export const VALID_BUTTON_TYPES = {
    iconButton: ['primary', 'secondary', 'ghost', 'ghostTertiary'],
    button: ['primary', 'secondary', 'outline', 'ghost', 'link']
};

export const VALID_BUTTON_SIZES = ['xsmall', 'small', 'medium', 'large'];
