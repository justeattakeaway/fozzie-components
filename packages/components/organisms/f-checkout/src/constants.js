export const CHECKOUT_METHOD_COLLECTION = 'collection';
export const CHECKOUT_METHOD_DELIVERY = 'delivery';

export const VALID_CHECKOUT_METHODS = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];

export const TENANT_MAP = {
    'en-GB': 'uk',
    'en-AU': 'au',
    'en-NZ': 'nz',
    'da-DK': 'dk',
    'es-ES': 'es',
    'en-IE': 'ie',
    'it-IT': 'it',
    'nb-NO': 'no'
};

export const VALIDATIONS = {
    address: 'addressValidations',
    guest: 'guestValidations'
};
