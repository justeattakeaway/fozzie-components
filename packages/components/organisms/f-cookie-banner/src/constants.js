export const CONSENT_COOKIE_NAME = 'je-cookieConsent';
export const LEGACY_CONSENT_COOKIE_NAME = 'je-banner_cookie';
