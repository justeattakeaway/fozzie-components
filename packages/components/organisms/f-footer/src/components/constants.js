export const confianzaUrl = 'https://www.confianzaonline.es/empresas/justeatspain.htm';
export const confianzaScreenReaderText = 'Accede a Confianza Online y comprueba nuestro sello de calidad en Internet';
