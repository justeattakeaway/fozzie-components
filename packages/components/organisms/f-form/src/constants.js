export const FORM_EVENTS = {
    fieldUpdated: 'updated',
    submitting: 'form-submitting'
};

export const PROP_VALIDATION_MESSAGES = {
    requiredFormFields: '`formFields` Not found in `formData` prop, please add an array of `formFields`',
    invalidFormFields: '`formFields` are invalid, please add an array of formFields',
    requiredNameProperty: 'Each field in `formFields` should contain "name" property',
    requiredTranslationsProperty: 'Each field in `formFields` should contain "translations" property',
    requiredLabelProperty: 'Each field in `formFields` should contain "translations.label" property'
};

export const DEFAULT_BUTTON_TEXT = 'Submit';
