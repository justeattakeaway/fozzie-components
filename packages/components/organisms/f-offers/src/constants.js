export const POSTCODE_EMPTY = 'POSTCODE_EMPTY';
export const POSTCODE_INVALID = 'POSTCODE_INVALID';
