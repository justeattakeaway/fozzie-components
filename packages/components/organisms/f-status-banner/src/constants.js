/* eslint-disable import/prefer-default-export */
export const BASE_HERO_URL = 'https://d30v2pzvrfyzpo.cloudfront.net/status-banner/';
