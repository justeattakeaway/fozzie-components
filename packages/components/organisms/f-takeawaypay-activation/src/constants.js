export const ACTIVATION_STATE_NONE = 'none';
export const ACTIVATION_STATE_AVAILABLE_LOGGED_IN = 'available_logged_in';
export const ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN = 'available_not_logged_in';
export const ACTIVATION_STATE_SUCCEEDED = 'succeeded';
export const ACTIVATION_STATE_FAILED = 'failed';
