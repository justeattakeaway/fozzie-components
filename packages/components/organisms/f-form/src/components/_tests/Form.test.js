import { shallowMount } from '@vue/test-utils';
import { PROP_VALIDATION_MESSAGES } from '../../constants';
import VForm from '../Form.vue';

const formData = {
    formFields: [
        {
            name: 'firstName',
            value: 'John',
            translations: {
                label: 'First Name'
            }
        },
        {
            name: 'lastName',
            value: 'Johnson',
            translations: {
                label: 'Last Name'
            }
        },
        {
            name: 'email',
            value: 'John.Johnson@gmail.com',
            translations: {
                label: 'Email Address'
            }
        }
    ],
    buttonText: 'Continue'
};

describe('Form', () => {
    let propsData;

    beforeEach(() => {
        propsData = { formData };
    });

    it('should be defined', () => {
        // Arrange & Act
        const wrapper = shallowMount(VForm, { propsData });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    describe('when `formData` prop is incorrect', () => {
        let spy;

        beforeEach(() => {
            spy = jest.spyOn(global.console, 'error').mockImplementation(() => { });
        });

        afterEach(() => {
            spy.mockRestore();
        });

        it('should throw an error when `formFields` property is missing from `formData`', () => {
            // Act & Assert
            expect(() => {
                shallowMount(VForm, { propsData: { formData: {} } });
            })
                .toThrowError(PROP_VALIDATION_MESSAGES.requiredFormFields);
        });

        it('should throw an error when `formFields` property is not an `Array`', () => {
            // Act & Assert
            expect(() => {
                shallowMount(VForm, { propsData: { formData: { formFields: {} } } });
            })
                .toThrowError(PROP_VALIDATION_MESSAGES.invalidFormFields);
        });

        it('should throw an error when field in `formFields` does not have `name` property', () => {
            // Act & Assert
            expect(() => {
                shallowMount(VForm, {
                    propsData: {
                        formData: {
                            formFields: [{
                                value: '',
                                translations: {
                                    label: 'Label Text'
                                }
                            }]
                        }
                    }
                });
            })
                .toThrowError(PROP_VALIDATION_MESSAGES.requiredNameProperty);
        });

        it('should throw an error when field in `formFields` does not have `translations` property', () => {
            // Act & Assert
            expect(() => {
                shallowMount(VForm, {
                    propsData: {
                        formData: {
                            formFields: [{
                                name: 'fieldName',
                                value: ''
                            }]
                        }
                    }
                });
            })
                .toThrowError(PROP_VALIDATION_MESSAGES.requiredTranslationsProperty);
        });

        it('should throw an error when field in `formFields` does not have `translations.label` property', () => {
            // Act & Assert
            expect(() => {
                shallowMount(VForm, {
                    propsData: {
                        formData: {
                            formFields: [{
                                name: 'fieldName',
                                value: '',
                                translations: {}
                            }]
                        }
                    }
                });
            })
                .toThrowError(PROP_VALIDATION_MESSAGES.requiredLabelProperty);
        });
    });

    describe('computed :: ', () => {
        describe('formFields :: ', () => {
            it('should return an object of `formData` keys and field values', () => {
                // Arrange & Act
                const wrapper = shallowMount(VForm, { propsData });

                const result = wrapper.vm.formFields;

                // Assert
                expect(result).toMatchSnapshot();
            });
        });
    });

    describe('methods :: ', () => {
        describe('updateField :: ', () => {
            it('should emit `updated` event with fieldName and field value', () => {
                // Arrange
                const wrapper = shallowMount(VForm, { propsData });
                const emitSpy = jest.spyOn(wrapper.vm, '$emit');
                const payload = { fieldName: 'firstName', value: 'Joe' };

                // Act
                wrapper.vm.updateField(payload);

                // Assert
                expect(emitSpy).toHaveBeenCalledWith('updated', payload);
            });
        });

        describe('onFormSubmit :: ', () => {
            it('should emit `form-submitting`', () => {
                // Arrange
                const wrapper = shallowMount(VForm, { propsData });

                // Act
                wrapper.vm.onFormSubmit();

                // Assert
                expect(wrapper.emitted('form-submitting').length).toBe(1);
            });
        });
    });
});
