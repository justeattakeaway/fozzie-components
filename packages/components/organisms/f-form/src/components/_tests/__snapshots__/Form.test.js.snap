// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form computed ::  formFieldData ::  should create an array of formFields with additional fieldData 1`] = `
Array [
  Object {
    "props": Object {
      "input-type": "text",
      "label-text": "First Name",
      "name": "first-name",
    },
    "translations": Object {
      "label": "First Name",
      "validationMessages": Object {
        "required": "Enter First Name",
      },
    },
    "validationMessages": Object {
      "required": "Enter First Name",
    },
    "value": "John",
  },
]
`;

exports[`Form computed ::  formFields ::  should return an object of \`formData\` keys and field values 1`] = `
Object {
  "firstName": "John",
}
`;
